diff -ru a/include/ap_fixed_special.h b/include/ap_fixed_special.h
--- a/include/ap_fixed_special.h	2025-09-16 11:47:35
+++ b/include/ap_fixed_special.h	2026-09-17 10:05:52
@@ -27,10 +27,15 @@
 #endif
 // FIXME AP_AUTOCC cannot handle many standard headers, so declare instead of
 // include.
-// #include <complex>
+#ifndef __SYNTHESIS__
+// Forward declaring std::complex is undefined behavior and does not work with libc++,
+// which declares it in an inline namespace (std::__1::complex).
+#include <complex>
+#else
 namespace std {
 template<typename _Tp> class complex;
 }
+#endif
 
 /*
   TODO: Modernize the code using C++11/C++14
diff -ru a/include/ap_int_special.h b/include/ap_int_special.h
--- a/include/ap_int_special.h	2025-09-16 11:47:35
+++ b/include/ap_int_special.h	2026-09-17 10:05:52
@@ -27,10 +27,15 @@
 #endif
 // FIXME AP_AUTOCC cannot handle many standard headers, so declare instead of
 // include.
-// #include <complex>
+#ifndef __SYNTHESIS__
+// Forward declaring std::complex is undefined behavior and does not work with libc++,
+// which declares it in an inline namespace (std::__1::complex).
+#include <complex>
+#else
 namespace std {
 template<typename _Tp> class complex;
 }
+#endif
 
 /*
   TODO: Modernize the code using C++11/C++14
