diff --git a/HLTrigger/Timer/src/processor_model.cc b/HLTrigger/Timer/src/processor_model.cc
index 1222e559a..c165a15b7 100644
--- a/HLTrigger/Timer/src/processor_model.cc
+++ b/HLTrigger/Timer/src/processor_model.cc
@@ -44,11 +44,13 @@ std::string read_processor_model() {
 
 #if BOOST_OS_BSD || BOOST_OS_MACOS
   // on BSD and OS X, read the processor  model via sysctlbyname("machdep.cpu.brand_string", ...)
-  std::string result;
-  size_t len;
-  sysctlbyname("machdep.cpu.brand_string", nullptr, &len, NULL, 0);
-  result.resize(len);
-  sysctlbyname("machdep.cpu.brand_string", result.data(), &len, NULL, 0);
+  size_t len = 0;
+  if (sysctlbyname("machdep.cpu.brand_string", nullptr, &len, NULL, 0) == 0 and len > 0) {
+    result.resize(len);
+    sysctlbyname("machdep.cpu.brand_string", result.data(), &len, NULL, 0);
+    // the length includes the terminating NUL
+    result.resize(len > 0 ? len - 1 : 0);
+  }
 #endif  // BOOST_OS_BSD || BOOST_OS_MACOS
 
   return result;
diff --git a/DQMServices/FileIO/plugins/DQMFileSaverPB.cc b/DQMServices/FileIO/plugins/DQMFileSaverPB.cc
index 8a96812c5..42ca7353c 100644
--- a/DQMServices/FileIO/plugins/DQMFileSaverPB.cc
+++ b/DQMServices/FileIO/plugins/DQMFileSaverPB.cc
@@ -350,7 +350,7 @@ int DQMFileSaverPB::getMaxCompressedSize(int bufferSize) const {
   return maxOutputSize;
 }
 
-ulong DQMFileSaverPB::compressME(const TBufferFile& buffer, int maxOutputSize, char* compression_output) const {
+unsigned long DQMFileSaverPB::compressME(const TBufferFile& buffer, int maxOutputSize, char* compression_output) const {
   z_stream deflateStream;
   deflateStream.zalloc = Z_NULL;
   deflateStream.zfree = Z_NULL;
diff --git a/DQMServices/FileIO/plugins/DQMFileSaverPB.h b/DQMServices/FileIO/plugins/DQMFileSaverPB.h
index 062dfa7b6..0da807362 100644
--- a/DQMServices/FileIO/plugins/DQMFileSaverPB.h
+++ b/DQMServices/FileIO/plugins/DQMFileSaverPB.h
@@ -43,7 +43,7 @@ namespace dqm {
 
   private:
     int getMaxCompressedSize(int bufferSize) const;
-    ulong compressME(const TBufferFile& buffer, int maxOutputSize, char* compression_output) const;
+    unsigned long compressME(const TBufferFile& buffer, int maxOutputSize, char* compression_output) const;
   };
 
 }  // namespace dqm
diff --git a/DQMServices/StreamerIO/plugins/DQMStreamerReader.cc b/DQMServices/StreamerIO/plugins/DQMStreamerReader.cc
index bfc4a442f..e4b87e573 100644
--- a/DQMServices/StreamerIO/plugins/DQMStreamerReader.cc
+++ b/DQMServices/StreamerIO/plugins/DQMStreamerReader.cc
@@ -17,6 +17,7 @@
 #include <fstream>
 #include <memory>
 #include <queue>
+#include <unistd.h>
 
 namespace dqmservices {
   using namespace edm::streamer;
diff --git a/Validation/Geometry/src/MaterialBudgetTree.cc b/Validation/Geometry/src/MaterialBudgetTree.cc
index 39d63ba46..480409c92 100644
--- a/Validation/Geometry/src/MaterialBudgetTree.cc
+++ b/Validation/Geometry/src/MaterialBudgetTree.cc
@@ -2,6 +2,7 @@
 #include "Validation/Geometry/interface/MaterialBudgetData.h"
 
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include <unistd.h>
 
 MaterialBudgetTree::MaterialBudgetTree(std::shared_ptr<MaterialBudgetData> data, const std::string &filename)
     : MaterialBudgetFormat(data) {
diff --git a/HeavyFlavorAnalysis/RecoDecay/src/BPHDecayMomentum.cc b/HeavyFlavorAnalysis/RecoDecay/src/BPHDecayMomentum.cc
index e94c562d5..2f56d877c 100644
--- a/HeavyFlavorAnalysis/RecoDecay/src/BPHDecayMomentum.cc
+++ b/HeavyFlavorAnalysis/RecoDecay/src/BPHDecayMomentum.cc
@@ -107,7 +107,7 @@ const reco::Candidate* BPHDecayMomentum::getDaug(const string& name) const {
     const BPHRecoCandidate* comp = getComp(name.substr(0, pos)).get();
     return (comp == nullptr ? nullptr : comp->getDaug(name.substr(pos + 1)));
   }
-  map<const string, const reco::Candidate*>::const_iterator iter = dMap.find(name);
+  auto iter = dMap.find(name);
   return (iter != dMap.end() ? iter->second : nullptr);
 }
 
@@ -119,7 +119,7 @@ BPHRecoConstCandPtr BPHDecayMomentum::getComp(const string& name) const {
     const BPHRecoCandidate* comp = getComp(name.substr(0, pos)).get();
     return (comp == nullptr ? nullptr : comp->getComp(name.substr(pos + 1)));
   }
-  map<const string, BPHRecoConstCandPtr>::const_iterator iter = cMap.find(name);
+  auto iter = cMap.find(name);
   return (iter != cMap.end() ? iter->second : nullptr);
 }
 
diff --git a/DQM/EcalMonitorClient/src/IntegrityClient.cc b/DQM/EcalMonitorClient/src/IntegrityClient.cc
index 281ff3590..cf9c96c77 100644
--- a/DQM/EcalMonitorClient/src/IntegrityClient.cc
+++ b/DQM/EcalMonitorClient/src/IntegrityClient.cc
@@ -67,7 +67,7 @@ namespace ecaldqm {
          chSItr.toNextChannel(GetElectronicsMap())) {
       DetId id(chSItr->getId());
 
-      EcalChannelStatusMap::const_iterator chIt(nullptr);
+      EcalChannelStatusMap::const_iterator chIt;
 
       // Set appropriate channel map (EB or EE)
       if (id.subdetId() == EcalBarrel) {
diff --git a/DQM/SiStripMonitorHardware/src/SiStripSpyUnpacker.cc b/DQM/SiStripMonitorHardware/src/SiStripSpyUnpacker.cc
index 412a6450b..b7544ea5f 100644
--- a/DQM/SiStripMonitorHardware/src/SiStripSpyUnpacker.cc
+++ b/DQM/SiStripMonitorHardware/src/SiStripSpyUnpacker.cc
@@ -20,7 +20,6 @@
 
 #include <algorithm>
 #include <vector>
-#include <ext/algorithm>
 
 using namespace std;
 
diff --git a/AnalysisDataFormats/TopObjects/src/printParticle.cc b/AnalysisDataFormats/TopObjects/src/printParticle.cc
index 75ab342d9..6614ded29 100644
--- a/AnalysisDataFormats/TopObjects/src/printParticle.cc
+++ b/AnalysisDataFormats/TopObjects/src/printParticle.cc
@@ -22,11 +22,11 @@ namespace ttevent {
       log << std::setw(15) << name << ": not available!\n";
       return;
     }
-    log << std::setprecision(3) << setiosflags(std::ios::fixed | std::ios::showpoint);
+    log << std::setprecision(3) << std::setiosflags(std::ios::fixed | std::ios::showpoint);
     log << std::setw(15) << name << ": " << std::setw(7) << cand->pt() << "; " << std::setw(7) << cand->eta() << "; "
-        << std::setw(7) << cand->phi() << "; " << resetiosflags(std::ios::fixed | std::ios::showpoint)
-        << setiosflags(std::ios::scientific) << std::setw(10) << cand->mass() << "\n";
-    log << resetiosflags(std::ios::scientific);
+        << std::setw(7) << cand->phi() << "; " << std::resetiosflags(std::ios::fixed | std::ios::showpoint)
+        << std::setiosflags(std::ios::scientific) << std::setw(10) << cand->mass() << "\n";
+    log << std::resetiosflags(std::ios::scientific);
   }
 
 }  // namespace ttevent
diff --git a/PhysicsTools/KinFitter/src/TKinFitter.cc b/PhysicsTools/KinFitter/src/TKinFitter.cc
index c67ccb632..b9011d1fa 100644
--- a/PhysicsTools/KinFitter/src/TKinFitter.cc
+++ b/PhysicsTools/KinFitter/src/TKinFitter.cc
@@ -1137,19 +1137,19 @@ void TKinFitter::print() {
     Int_t nParP = particle->getNPar();
     const TMatrixD* par = particle->getParCurr();
     const TMatrixD* covP = particle->getCovMatrix();
-    log << std::setw(3) << setiosflags(std::ios::right) << iP;
-    log << std::setw(15) << setiosflags(std::ios::right) << particle->GetName();
+    log << std::setw(3) << std::setiosflags(std::ios::right) << iP;
+    log << std::setw(15) << std::setiosflags(std::ios::right) << particle->GetName();
     log << std::setw(3) << " ";
     for (int iPar = 0; iPar < nParP; iPar++) {
       if (iPar > 0) {
-        log << setiosflags(std::ios::right) << std::setw(21) << " ";
+        log << std::setiosflags(std::ios::right) << std::setw(21) << " ";
       }
       TString colstr = "";
       colstr += parIndex;
       colstr += ":";
       log << std::setw(4) << colstr;
       log << std::setw(2) << " ";
-      log << setiosflags(std::ios::left) << setiosflags(std::ios::scientific) << std::setprecision(3);
+      log << std::setiosflags(std::ios::left) << std::setiosflags(std::ios::scientific) << std::setprecision(3);
       log << std::setw(15) << (*par)(iPar, 0);
       if (_nbIter > 0 && _status < 10) {
         log << std::setw(15) << TMath::Sqrt(_yaVFit(iPar, iPar));
@@ -1169,19 +1169,19 @@ void TKinFitter::print() {
     TAbsFitParticle* particle = _unmeasParticles[iP];
     Int_t nParP = particle->getNPar();
     const TMatrixD* par = particle->getParCurr();
-    log << std::setw(3) << setiosflags(std::ios::right) << iP;
+    log << std::setw(3) << std::setiosflags(std::ios::right) << iP;
     log << std::setw(15) << particle->GetName();
     log << std::setw(3) << " ";
     for (int iPar = 0; iPar < nParP; iPar++) {
       if (iPar > 0) {
-        log << setiosflags(std::ios::right) << std::setw(21) << " ";
+        log << std::setiosflags(std::ios::right) << std::setw(21) << " ";
       }
       TString colstr = "";
       colstr += parIndex;
       colstr += ":";
       log << std::setw(4) << colstr;
       log << std::setw(2) << " ";
-      log << setiosflags(std::ios::left) << setiosflags(std::ios::scientific) << std::setprecision(3);
+      log << std::setiosflags(std::ios::left) << std::setiosflags(std::ios::scientific) << std::setprecision(3);
       log << std::setw(15) << (*par)(iPar, 0);
       if (_nbIter > 0 && _status < 10) {
         log << std::setw(15) << TMath::Sqrt(_yaVFit(iPar + _nParB, iPar + _nParB));
diff --git a/DetectorDescription/OfflineDBLoader/plugins/OutputDDToDDL.cc b/DetectorDescription/OfflineDBLoader/plugins/OutputDDToDDL.cc
index 831f2af7b..0f8a2d8e6 100644
--- a/DetectorDescription/OfflineDBLoader/plugins/OutputDDToDDL.cc
+++ b/DetectorDescription/OfflineDBLoader/plugins/OutputDDToDDL.cc
@@ -216,8 +216,7 @@ void OutputDDToDDL::beginRun(const edm::Run&, edm::EventSetup const& es) {
   (*m_xos) << "</LogicalPartSection>" << std::endl;
 
   (*m_xos) << std::fixed << std::setprecision(18);
-  std::map<DDsvalues_type, std::set<const DDPartSelection*> >::const_iterator mit(specStore.begin()),
-      mend(specStore.end());
+  auto mit = specStore.cbegin(), mend = specStore.cend();
   (*m_xos) << "<SpecParSection label=\"" << ns_ << "\">" << std::endl;
   for (; mit != mend; ++mit) {
     out.specpar(*mit, *m_xos);
diff --git a/DetectorDescription/OfflineDBLoader/plugins/OutputMagneticFieldDDToDDL.cc b/DetectorDescription/OfflineDBLoader/plugins/OutputMagneticFieldDDToDDL.cc
index fd3b07e88..019014922 100644
--- a/DetectorDescription/OfflineDBLoader/plugins/OutputMagneticFieldDDToDDL.cc
+++ b/DetectorDescription/OfflineDBLoader/plugins/OutputMagneticFieldDDToDDL.cc
@@ -216,8 +216,7 @@ void OutputMagneticFieldDDToDDL::beginRun(const edm::Run&, edm::EventSetup const
   (*m_xos) << "</LogicalPartSection>\n";
 
   (*m_xos) << std::fixed << std::setprecision(18);
-  std::map<DDsvalues_type, std::set<const DDPartSelection*> >::const_iterator mit(specStore.begin()),
-      mend(specStore.end());
+  auto mit = specStore.cbegin(), mend = specStore.cend();
   (*m_xos) << "<SpecParSection label=\"" << ns_ << "\">\n";
   for (; mit != mend; ++mit) {
     out.specpar(*mit, *m_xos);
diff --git a/Alignment/OfflineValidation/src/Trend.cc b/Alignment/OfflineValidation/src/Trend.cc
index c2e79c1d1..f9aa70f76 100644
--- a/Alignment/OfflineValidation/src/Trend.cc
+++ b/Alignment/OfflineValidation/src/Trend.cc
@@ -26,7 +26,7 @@ Run2Lumi::Run2Lumi(fs::path file, int first, int last, float convertUnit)
 
   assert(fs::exists(file));
 
-  ifstream f(file);
+  ifstream f(file.string());
   int run;
   while (f >> run >> runs[run])
     ;
diff --git a/DQM/TrackerRemapper/bin/BuildFile.xml b/DQM/TrackerRemapper/bin/BuildFile.xml
index 5576210c2..eb88a3447 100644
--- a/DQM/TrackerRemapper/bin/BuildFile.xml
+++ b/DQM/TrackerRemapper/bin/BuildFile.xml
@@ -4,7 +4,6 @@
 <use name="rootcore"/>
 <use name="rootgraphics"/>
 <use name="rootmath"/>
-<lib name="stdc++fs" />
 <bin name="printPixelTrackerMap" file="printPixelTrackerMap.cc" />
 <bin name="printPixelROCsMap" file="printPixelROCsMap.cc" />
 <bin name="printPixelLayersDisksMap" file="printPixelLayersDisksMap.cc" />
